### PR TITLE
DateTimeParseException should be an exception too

### PR DESCRIPTION
**Pull request by rcsilva83**
It's the same case of NumberFormatException, ParseException and MalformedURLException.

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/CatchUsesExceptionWithContextCheck.java [hunk 1]
```diff
@@ -19,9 +19,11 @@
  */
 package org.sonar.java.checks;
 
-import com.google.common.base.Splitter;
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Lists;
+import java.util.ArrayDeque;
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/CatchUsesExceptionWithContextCheck.java]: Please don't change order of imports. These are useless changes. Did you used had a look at our coding style? https://github.com/SonarSource/sonar-developer-toolset#code-style

File: java-checks/src/test/files/checks/CatchUsesExceptionWithContextCheck.java [hunk 2]
```diff
@@ -93,6 +93,7 @@ private void h() {
       /* do nothing */
     } catch (ParseException e) {                 // Compliant
     } catch (MalformedURLException e) {          // Compliant
+    } catch (DateTimeParseException e) {          // Compliant
```
Comment (m-g-sonar) [inline java-checks/src/test/files/checks/CatchUsesExceptionWithContextCheck.java]: Use the fully qualified name for the exception (`java.time.format.DateTimeParseException`). As you didn't added the exception to the imports, the analyzer won't be able to resolve its type and will consider it as unknown... and unknown exception types do not raise issues.

File: java-checks/src/test/files/checks/CatchUsesExceptionWithContextCheck.java [hunk 3]
```diff
@@ -125,6 +126,7 @@ private void h() {
     } catch (java.lang.NumberFormatException e) { // Compliant
     } catch (java.net.MalformedURLException e) {  // Compliant
     } catch (java.text.ParseException e) {        // Compliant
+    } catch (DateTimeParseException e) {          // Compliant
```
Comment (m-g-sonar) [inline java-checks/src/test/files/checks/CatchUsesExceptionWithContextCheck.java]: Same remark for the fully qualified name.

**Comment by rcsilva83** [comment]
Thanks for your feedback, @m-g-sonar!

I accidentally closed this PR and created a new one changing the code according to your considerations: #1031

**Pull request closed without merging**